test/integration/System.t.sol

```diff
@@ -0,0 +1,632 @@
+        assertEq(timelock.pauseGuardian(), guardian, "incorrect pauser");
+        assertEq(
+            timelock.pauseDuration(), PAUSE_DURATION, "incorrect pause duration"
+        );
+    }
+
+    ///
+    /// 2. Queue and execute a transaction in the Gnosis Safe to perform the following actions:
+    ///  - initialize configuration with the timelock address, and allowed time ranges and their corresponding allowed days
+    ///  - add the guard to the Safe
+    ///  - add the Timelock as a Safe module
+
+    ///
+    /// construction of initial setup call in this system:
+    ///  1. call to SafeRestriction contract to initialize configuration
+    ///  with the timelock address, and allowed time ranges and their
+    ///  corresponding allowed days.
+    /// 2. call to `setGuard` with the address of the time-restriction contract on the safe
+    /// 3. call `enableModule` with the address of the timelock on the safe
+    ///
+    /// Notes:
+    ///   This should be wrapped in a single call to the Safe contract.
+    ///   Use multicall to execute the calls in a single transaction.
+    ///
+    ///
+    /// construction of all calls within this system outside of setup:
+    ///    safe calls timelock, timelock calls external contracts
+    ///    encoding:
+    ///       1. gather array of addresses, values and bytes for the calls
+    ///       2. encode the array of calls to call the scheduleBatch function on the timelock
+    ///       3. encode this data to call the Safe contract
+    ///
+    ///
+    function testInitializeContract() public {
+        address[] memory calls = new address[](3);
+        calls[0] = address(restricted);
+        calls[1] = address(safe);
+        calls[2] = address(safe);
+
+        bytes[] memory calldatas = new bytes[](3);
+
+        {
+            uint8[] memory allowedDays = new uint8[](5);
+            allowedDays[0] = 1;
+            allowedDays[1] = 2;
+            allowedDays[2] = 3;
+            allowedDays[3] = 4;
+            allowedDays[4] = 5;
+
+            TimeRestricted.TimeRange[] memory ranges =
+                new TimeRestricted.TimeRange[](5);
+
+            ranges[0] = TimeRestricted.TimeRange(10, 11);
+            ranges[1] = TimeRestricted.TimeRange(10, 11);
+            ranges[2] = TimeRestricted.TimeRange(12, 13);
+            ranges[3] = TimeRestricted.TimeRange(10, 14);
+            ranges[4] = TimeRestricted.TimeRange(11, 13);
+
+            calldatas[0] = abi.encodeWithSelector(
+                restricted.initializeConfiguration.selector,
+                address(timelock),
+                ranges,
+                allowedDays
+            );
+        }
+
+        calldatas[1] = abi.encodeWithSelector(
+            GuardManager.setGuard.selector, address(restricted)
+        );
+
+        calldatas[2] = abi.encodeWithSelector(
+            ModuleManager.enableModule.selector, address(timelock)
+        );
+
+        call3.Call3[] memory calls3 = new call3.Call3[](3);
```

Comment:
If we move this declaration to the first line of the function, we won't need to declare the temporary variables above like `calls` and `calldatas`.